File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/validation/validator/UniqueCollectValidator.java
package com.windea.demo.cloudcollect.core.validation.validator;

import com.windea.demo.cloudcollect.core.domain.entity.Collect;
import com.windea.demo.cloudcollect.core.service.CollectService;
import com.windea.demo.cloudcollect.core.validation.annotation.UniqueCollect;
import org.springframework.beans.factory.annotation.Autowired;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

/**
 * 唯一收藏的校验器。
 */
public class UniqueCollectValidator implements ConstraintValidator<UniqueCollect, Collect> {
	@Autowired private CollectService service;

	@Override
	public boolean isValid(Collect value, ConstraintValidatorContext context) {
		return !service.exists(value);
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/validation/validator/UniqueCollectValidator.kt
package com.windea.demo.cloudcollect.core.validation.validator

import com.windea.demo.cloudcollect.core.domain.entity.*
import com.windea.demo.cloudcollect.core.service.*
import com.windea.demo.cloudcollect.core.validation.annotation.*
import org.springframework.beans.factory.annotation.*
import javax.validation.*

/**唯一收藏的校验器。*/
open class UniqueCollectValidator : ConstraintValidator<UniqueCollect, Collect> {
	@Autowired private lateinit var collectService: CollectService
	
	override fun isValid(value: Collect, context: ConstraintValidatorContext): Boolean {
		return !collectService.existsByNameAndUser(value.name, value.user)
	}
}
